### ISSUE TITLE
Files during accidental mass deletion lost and not found even in Trashbin!

### ISSUE THREAD

**Issue by brozkeff** [owncloud/core#12129]
### Steps to reproduce
1. Install OC server 7.0.3 on your own linux VPS
2. create a few users, upload a few GB of data from user A with unlimited quota
3. share dozens of individual files and directories of 1000s of files and a few GBs in size to other users
4. let them connect on another computers with Windows 7 to start downloading the shared files
5. as user A log to web interface, select "Files shared with others" and instad of un-sharing, start clicking the trashbin icon to delete quickly one file/directory after another (I was thinking I am unsharing files but actually I was deleting them even from myself, and the changes immediately started proppagating everywhere)
6. realize you did a mistake, and files are being deleted instead of just being unshared, go to trashbin to restore it all back, start restoring what you find there but eventually some folders (for me it was the biggest of them) do not show in the recycle bin, nor they can be found on the server physically... the clients on the computers are also already deleting all the files
7. stop this madness by reverting whole VPS to the last snapshot

### Expected behaviour
a) When being in the tab "shared by me  to others" the trash bin should be replaced with Unshare button, and deleting when not in the main filelist should not be allowed at all.
The behaviour when being in the tab "shared to me from others" is that clicking recycle bin unshares it but it remains on the file owner's account

b) even if such heavy deleting/restoring happens and multiple clients start syncing, there must never happen that files and directories vanish completely - all must be still found in trashbin in any circumstances and files must be on the server physically in the directory structure somewhere

### Actual behaviour
Lot of files and folders vanished completely from the filesystem, the OC trashbin etc. See Steps to reproduce

### Server configuration
**Operating system**:
Ubuntu server 14.04

**Web server:**
Apache 2.4

**Database:**
MySQL 5.5.35

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
7.0.3

**Updated from an older ownCloud or fresh install:**
originally 6.0.4
then gradually to 7.0 RC, and each version

**List of activated apps:**
Activity
Calendar
Contacts
Deleted Files
Docments
Pictures
Share files
Text Editor
Versions

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password and passwordsalt)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no
no

### Client configuration
**Browser:**
Firefox 33

**Operating system:**
GNU/Linux Mint 17 Cinnamon

#### Web server error log
```
i will send on demand by email
```

#### ownCloud log (data/owncloud.log)
just part of the log from the administration interface
```
Debug 	webdav 	Sabre\DAV\Exception\NotAuthenticated: No basic authentication headers were found 	November 12, 2014 12:43:38
Debug 	OC\Files\Cache\Scanner 	!!! No reuse of etag for 'files_versions/MM-GRAFIKA' !!! cache: Array ( [fileid] => 74446 [storage] => home::martin.malec [path] => files_versions/MM-GRAFIKA [parent] => 27994 [name] => MM-GRAFIKA [mimetype] => httpd/unix-directory [mimepart] => httpd [size] => -1 [mtime] => 1415650720 [storage_mtime] => 1415650720 [encrypted] => [unencrypted_size] => 0 [etag] => 546336339742d [permissions] => 31 ) data: Array ( [mimetype] => httpd/unix-directory [mtime] => 1415789726 [size] => -1 [etag] => 546347cab0545 [storage_mtime] => 1415789726 [permissions] => 31 [parent] => 27994 ) 	November 12, 2014 12:43:06
Warning 	core 	Login failed: 'martin.malec' (Remote IP: '81.19.3.71', X-Forwarded-For: '') 	November 12, 2014 12:43:00
Warning	core	Login failed: 'martin.malec' (Remote IP: '81.19.3.71', X-Forwarded-For: '')	November 12, 2014 12:43:00
Error	PHP	apc_store(): GC cache entry 'oc1fdcd11858/AutoloaderOC\Log\ErrorHandler' was on gc-list for 3612 seconds at /var/www/owncloud/lib/private/memcache/apc.php#21	November 12, 2014 12:42:54
Error	PHP	apc_store(): GC cache entry 'oc1fdcd11858/AutoloaderOC\Files\Stream\StaticStream' was on gc-list for 4104 seconds at /var/www/owncloud/lib/private/memcache/apc.php#21	November 12, 2014 12:37:46
Debug	carddav	Sabre\DAV\Exception\NotAuthenticated: No basic authentication headers were found	November 12, 2014 12:37:46
Debug	carddav	Sabre\DAV\Exception\NotAuthenticated: No basic authentication headers were found	November 12, 2014 12:19:42
Error	PHP	apc_store(): GC cache entry 'oc1fdcd11858/AutoloaderOC\Files\Storage\Loader' was on gc-list for 3630 seconds at /var/www/owncloud/lib/private/memcache/apc.php#21	November 12, 2014 12:04:43
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/_znacky materialu/označení, LDPE,HDPE, PET.ai could not be located	November 12, 2014 12:03:46
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/SPORTOVNI-PRACI.pdf could not be located	November 12, 2014 12:03:46
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/SILICE KOLECKA.pdf could not be located	November 12, 2014 12:03:45
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/PRACI GELY.pdf could not be located	November 12, 2014 12:03:45
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/PRACI GEL BEZ novy.pdf could not be located	November 12, 2014 12:03:44
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/MYCKA.pdf could not be located	November 12, 2014 12:03:41
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/LAHVE OSTATNI.pdf could not be located	November 12, 2014 12:03:40
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/GEL NADOBI.pdf could not be located	November 12, 2014 12:03:39
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/AVIVAZ.pdf could not be located	November 12, 2014 12:03:39
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/SLEEVY/32417x.eps could not be located	November 12, 2014 12:03:38
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/PAPIROVE/LETAKY/SKLADACKY/LETAKY-A4-CZ-SK-SE_.pdf could not be located	November 12, 2014 12:03:36
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/PAPIROVE/LETAKY/SKLADACKY/LETAKY-A4-CZ-SK-SE.pdf could not be located	November 12, 2014 12:03:36
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/PAPIROVE/LETAKY/SKLADACKY/LETAK SK.pdf could not be located	November 12, 2014 12:03:34
Debug	webdav	Sabre\DAV\Exception\NotFound: File with name MM-GRAFIKA/ETIKETY/misc/Y&B 2010-2013/PAPIROVE/LETAKY/SKLADACKY/LETAK SE.pdf could not be located	November 12, 2014 12:03:34
```

Actually I had 30 min old ZFS snapshot of the OpenVZ container where the company owncloud runs so I am now reverting to that snapshot. But otherwise it would be a major issue.

**Comment by PVince81** [comment]
The only explanation I see for files not appearing in the trashbin is that they got expired due to lack of space. But if there was enough space (and no quota) then this unlikely.

**Comment by MorrisJobke** [comment]
This should be solved with the high level file locking (implemented in 8.1 and activated in 8.2). Therefore I will close this ticket here.
